pymeasure/display/windows.py

```diff
@@ -522,6 +522,11 @@ def finished(self, experiment):
             self.abort_button.setEnabled(False)
             self.browser_widget.clear_button.setEnabled(True)
 
+    @property
+    def directory(self):
+        if not self.directory_input:
+            raise Exception("No directory input in the ManagedWindow")
```

Comment:
Judging purely from the exception message, a `ValueError` seems an appropriate exception type here.

Reply:
I have to say, I just copied the exception of the function make procedure. Do I change to this exception ?

Reply:
I see. In principle it's desirable to make the exceptions one throws _as specific as possible_, choosing basically from [this list](https://docs.python.org/3.7/library/exceptions.html?highlight=exception#concrete-exceptions). This has the benefit that whoever consumes your code and deals with these exceptions has an easier time filtering for the correct errors. You can also define your own exception hierarchy in code, but I don't think that's necessary here.
Yes, you would replace `Exception` with `ValueError` (or another more fitting one if you find one in the list, this  is just the one that came to my mind).

Reply:
Done.